elintegrostartapp/grails-app/services/com/elintegro/translator/TranslatorService.groovy

```diff
@@ -80,8 +80,18 @@ class TranslatorService {
         ine.close();
         return response.toString();
     }
+    boolean validateSourceFile(params, String projectName,String fileName) {
+        String language = "English"
+        def file = Holders.grailsApplication.config.images.storageLocation + "/images/" + "${projectName}" + "/" + fileName
+        def lines = file.readLines()
+        lines.each { record ->
+            String[] keyValue = record.split("=")
+            if (keyValue.length == 2) {
+                String key = keyValue[0]
+                println(key)
```

Comment:
Where is the validation for English?

The validation should be a regualr expression to validate that in this key all symbols are either English letters or special symbols like ., -, $, etc. Please take time to learn regexp. Use the following website to try and test your regex: 
https://regexr.com/
  Refrer to this code (it an example):
                                
				Pattern regExPattern = Pattern.compile(/[A-Z,\\.,\\-]/, , Pattern.CASE_INSENSITIVE);
				Matcher matcher = regExPattern.matcher(key);


				if(!matcher.find()) { //Not English!
                                            return false
				}

It is pseudocode!